### PR TITLE
Feat/city weather page

### PR DESCRIPTION
**Pull request by danielcurtin**
Added an entire page layout for city weather display

Added a background image to home page (I know, the branch)

Added logic and functionality for dynamic page rendering - works without the use of mock/static data

Added a button to go back to the home page in the form of the logo on a header

It's all decently responsive, but more checks and tests should be done to ensure better UX at lower screens

Refactored Header component to be reusable

Still some comments and console logs that need cleaned up but aside from the daily limit, mock-data really only kept for future cypress testing.

### CODE REVIEW

File: src/components/App/App.css [hunk 1]
```diff
@@ -10,9 +6,33 @@ main {
...
+  opacity: 0.5;
+  position: absolute;
+  z-index: -1;
+}
+
+@media screen and (max-width: 89rem) {
+  .home-bg {
+    height: 150%;
+    width: 150%;
+  }
```
Comment (jfogiato) [inline src/components/App/App.css:26]: I hadn't thought of having media queries broken out by component, but I like this flow.

File: src/components/App/App.js [hunk 2]
```diff
@@ -28,13 +32,18 @@ const App = () => {
...
     .then(res => {
       setWeather(res.data.values);
+      getWeatherType();
       history.push(`/city/${location}`)
     })
     .catch(() => setError(true));
   };
 
+  const getWeatherType = () => {
+    setWeatherType(weatherCodes[weather.weatherCode])
```
Comment (jfogiato) [inline src/components/App/App.js:42]: Appreciate the structure of this object - easy to reference the property you need via key

File: src/components/City/City.js [hunk 3]
```diff
@@ -2,11 +2,37 @@ import React from "react";
...
+    } else if (type.includes('Cloudy')) {
+      return cloudBg;
+    } else if (type.includes('Snow') || type.includes('Ice') || type.includes('Flurries')) {
+      return snowBg;
+    } else if (type.includes('Rain') || type.includes('Drizzle')) {
+      return rainBg;
+    } else if (type.includes('Thunderstorm')) {
+      return stormBg;
+    } else if (type.includes('Fog')) {
+      return fogBg;
```
Comment (jfogiato) [inline src/components/City/City.js:28]: I remember you mentioning not being super happy with this piece of code...

I haven't dug into this too much for a proof of concept, so I may be missing a crucial piece, but how about if you built out an object with your types and their associated backgrounds, then referenced that property by it's type key to get back the background you needed. eg:


`const bkgdObk = {`
`  Cloudy: cloudbkgd,`
`  Sunny: sunbkgd,`
`etc....`
`}`

`url(${bkgdObj[type]})`


Let me know if that makes sense and/or if you wanna chat through it!

File: src/components/Search/Search.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ const Search = ({ searchLocation }) => {
     <form role="search" className="search">
       <input type="text" className="search-bar" id="searchBar" value={search} onChange={event => setSearch(event.target.value)}></input>
       <button type="submit" className="search-button fa-solid fa-magnifying-glass" onClick={() => searchLocation(search)}></button>
-      <label htmlFor="searchBar">Enter U.S. City or Zip Code</label>
+      <label htmlFor="searchBar">Enter City or Zip Code</label>
```
Comment (jfogiato) [inline src/components/Search/Search.js:11]: 🌎

File: src/components/Weather/Weather.js [hunk 5]
```diff
@@ -1,22 +1,64 @@
...
+    } else if (temp < 20) {
+      return 'It\'s pretty cold for the average dog - unless they prefer temperatures like this, limit to in & out until it warms up.'
+    } else if (temp < 45) {
+      return 'Small or Thin Coat dogs might get cold. Maybe only 15 minutes?';
+    } else if (type.includes('Sunny') || type.includes('Clear')) {
+      return 'It\'s nice out! Remember to stay hydrated!';
+    } else if (type.includes('Snow') || type.includes('Flurries') || type.includes('Ice') || type.includes('Rain') || type.includes('Drizzle') || type.includes('Thunderstorm')) {
+      return 'It\'s not all that nice out... but it\'s up to you.';
+    } else if (type.includes('Fog')) {
+      return 'If you can see your surroundings, go for it!';
```
Comment (jfogiato) [inline src/components/Weather/Weather.js:44]: Potentially same with these bits - adding an object and accessing based on type

**Review by jfogiato — COMMENTED** [review]
Great work so far!! Super easy to read and straightforward flow. I added a bit about a potential solution for your length if/else statements, but maybe it won't be a good fit for the problem. Let me know your thoughts!

**Comment by danielcurtin** [comment]
Thanks! I like that idea of the object for the conditionals, a little hesitant off the top of my head because some of the types are more than one ... type, I guess (ie. "Clear, Sunny" is a possible outcome) but if I just make the object that specific then that could be a great substitute.

I'm glad someone likes the broken out media queries, because I can't stand 'em. For some reason I just assumed that was how it was supposed to be done and didn't even consider otherwise haha.

**Pull request merged**